src/ExpectedMessage.ts

```diff
@@ -136,4 +141,30 @@ export class ExpectedMessage {
 
         this.internalExpectation.expect(expectedResponseCollectionAsIMessage).toDeeplyInclude(outgoingMessage);
     }
+
+    /**
+     *  Verfy the incoming message with custom test defined by tester
+     *  If the function that tester defined return an error, make the test break
+     *  If the function return anything else, the test is considered as good
+     *  I've tryed to use promise as parameter, but in a promise we change scope, so the assert doesn't work
+     * @param {IMessage} outgoingMessage outgoing message being compared
+     */
+    private deepMatchCheckWithFunction(outgoingMessage: IMessage): void {
+        const functionCollection: Function[] = this.expectedResponseCollection as Function[];
+        let errorString = '';
+        let success = false;
+        functionCollection.forEach((func: Function) => {
```

Comment:
Have you tried bluebird `Promise.map` to allow for promise cases?

Reply:
Yes, but we are in the promise context, the  "this.internalExpectation.expect(success, error).toBeTrue();" create an error but test still a success